src/test/server/server_test_runner.cpp

```diff
@@ -125,4 +126,33 @@ TEST_F(ServerTestRunner, TestPreparedStatement) {
   EXPECT_EQ(result2.size(), 2u);
 }
 
+TEST_F(ServerTestRunner, TestParallelConnections) {
+  // This test is by no means perfect, as it can show flaky behaviour. But it is rather hard to get reliable tests with
```

Comment:
What does "flaky behavior" mean? False negatives is somewhat ok for this type of stress tests, false positives not.

Reply:
The test can miss the bug (i.e. false negative) but it can not fail when there is no bug. It just sends 100 parallel connections and if that fails, there is a bug :)

Reply:
Could you clarify that in the comment?